### PR TITLE
fontconfig: Remove patch for freetype libtool version number

### PR DESCRIPTION
**Pull request by jwillikers**
This patch is a bad workaround for improper versioning in the freetype package. This is fixed in the freetype package itself in #22974.

In addition, this fixes the following error, which can be seen when running the test package for the Meson variant of the package:

```
Fontconfig error: Cannot load default config file: No such file: (null)
```

This is fixed by installing the fontconfig configuration file.
It was being installed to the wrong location in the Meson package previously.
Additionally, don't set `FONTCONFIG_FILE`, but instead append to `FONTCONFIG_PATH` only.
This works and makes it easier for consumers to opt for the system font directory.
See #5782.

Requires:

- [x] #22974

### CODE REVIEW

**Review by ericLemanissier — APPROVED** [review]
this PR seems to be urgent. LGTM

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8799846282/attempts/1)

**Comment by jellespijker** [comment]
thanks for helping out @ericLemanissier

**Review by jellespijker — APPROVED** [review]
lgtm

**Comment by jellespijker** [comment]
> I told the team yesterday to merge this immediately after the freetype one. I'll update it to at least get it building.

Out of curiosity, did you guys ever experimented with GitHub merge queue's? It's on my bucket list to investigate the usefulness for our project. It seems to be helpful for these kind of stacked PR's

File: recipes/fontconfig/meson/conanfile.py [hunk 1]
```diff
@@ -75,15 +74,14 @@ def generate(self):
             "doc": "disabled",
             "nls": "disabled",
             "tests": "disabled",
-            "tools": "disabled"
+            "tools": "disabled",
+            "sysconfdir": os.path.join(self.package_folder, "res", "etc"),
+            "datadir": os.path.join(self.package_folder, "res", "share"),
```
Comment (ericLemanissier) [inline recipes/fontconfig/meson/conanfile.py:79]:
```suggestion
            "sysconfdir": os.path.join("res", "etc"),
            "datadir": os.path.join("res", "share"),
```
according to https://c3i.jfrog.io/c3i/misc/logs/pr/23010/8-linux-gcc-legacy/fontconfig/2.14.2//1dd5f947341f0c35b104c3cbc22073e69527248e-build.txt
these two paths seem to be relative
see also https://gitlab.freedesktop.org/fontconfig/fontconfig/-/blob/2.14.2/meson.build?ref_type=tags#L291 and 
https://gitlab.freedesktop.org/fontconfig/fontconfig/-/blob/2.14.2/meson.build?ref_type=tags#L301

Comment (jwillikers) [reply recipes/fontconfig/meson/conanfile.py:79]: Oh yeah, I'll need to update the version requirement too.

**Review by ericLemanissier — APPROVED** [review]

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8799846282/attempts/2)

**Review by jellespijker — APPROVED** [review]
LGTM

**Review by franramirez688 — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Comment by jellespijker** [comment]
Thanks everyone on behalf of team Cura for giving this PR priority.

It's appreciated :hearth: !

File: recipes/fontconfig/meson/conanfile.py [hunk 2]
```diff
@@ -3,16 +3,15 @@
...
 )
 from conan.tools.gnu import PkgConfigDeps
 from conan.tools.layout import basic_layout
 from conan.tools.meson import Meson, MesonToolchain
-from conan.tools.scm import Version
 
 import os
 
-required_conan_version = ">=1.53.0"
+required_conan_version = ">=1.64.0 <2 || >=2.2.0"
```
Comment (valgur) [inline recipes/fontconfig/meson/conanfile.py:14]: @jwillikers Why was this massive bump to the required Conan version necessary?

Related to https://github.com/conan-io/conan/issues/16140

**Pull request merged**